Framework/Algorithms/inc/MantidAlgorithms/WorkspaceBoundingBox.h

```diff
@@ -0,0 +1,66 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "MantidAPI/MatrixWorkspace_fwd.h"
+#include "MantidKernel/Logger.h"
+
+namespace Mantid {
+namespace Algorithms {
+
+/* This is a simple class originally intended for use solely with FindCenterOfMassPosition2.cpp
+ *
+ */
+class WorkspaceBoundingBox {
+public:
+  WorkspaceBoundingBox(API::MatrixWorkspace_sptr workspace, Kernel::Logger &g_log);
+  WorkspaceBoundingBox(Kernel::Logger &g_log);
```

Comment:
Is it necessary to passing the logger this way?  You should be able to define a logger in the namespace 

```c++
namespace Mantid {
namespace Algorithms {
    namespace {
    // static logger
    Logger g_log("WorkspaceBoundingBox");
    } // namespace
...
}
}
```

Reply:
This helper class is tied pretty strongly to FindCenterOfMass2, as such I thought it should just continue to use the FindCenterOfMass2 logger as not to confuse the log readers.  I can change it if you still think it is necessary.

Reply:
I think it's better to let users know where the messages actually coming from.
Please make `WorkspaceBoundingBox` using its own logger :pray:

Reply:
alrighty, added